server/src/main/java/org/elasticsearch/index/IndexVersion.java

```diff
@@ -117,7 +117,7 @@ private static IndexVersion registerIndexVersion(int id, Version luceneVersion,
     public static final IndexVersion V_8_9_0 = registerIndexVersion(8_09_00_99, Version.LUCENE_9_7_0, "32f6dbab-cc24-4f5b-87b5-015a848480d9");
     public static final IndexVersion V_8_9_1 = registerIndexVersion(8_09_01_99, Version.LUCENE_9_7_0, "955a80ac-f70c-40a5-9399-1d8a1e5d342d");
     public static final IndexVersion V_8_10_0 = registerIndexVersion(8_10_00_99, Version.LUCENE_9_7_0, "2e107286-12ad-4c51-9a6f-f8943663b6e7");
-    public static final IndexVersion V_8_11_0 = registerIndexVersion(8_11_00_99, Version.LUCENE_9_7_0, "f08382c0-06ab-41f4-a56a-cf5397275627");
+    public static final IndexVersion V_8_11_0 = registerIndexVersion(8_11_00_99, Version.LUCENE_9_8_0, "f08382c0-06ab-41f4-a56a-cf5397275627");
```

Comment:
This doesn't seem right. We shouldn't change existing definitions. It's odd that we have switched to independent index versions, yet we have a constant for the in-flight 8.11 version. I think uses of this constant need to be converted to using 8_500_000 and this constant removed. That can happen as a followup, but I think it is incorrect to change the lucene version associated with this constant.

Reply:
I was wondering about this same bit too, and I thought that it may be ok just because 8.11 is unreleased, and the new lucene release is in fact the lucene version it is going to depend on after all. I trust that it is incorrect, I did not get entirely if I need to do something else in my PR, hold on some other upstream change, or merge as-is but we'll need to adjust things as a follow-up.

Reply:
I think it doesn't make sense to have an IndexVersion for 8.11.0. We should remove this, and change references to use 8_500_000 - so we don't have lucene version going from 9.7 -> 9.8 -> 9.7 -> 9.8

Reply:
I think leave 8.11.0 at 9.7 for now, and we can remove it in a subsequent PR

Reply:
https://github.com/elastic/elasticsearch/pull/100188

Reply:
ops, for some reason I missed your comment. I merged, and left 8.11 with 9.8. I do think that that is the most correct things to do, because after the merge, 8.11 is based on 9.8 and not 9.7. Are you taking care of the needed follow-up?